src/crafting.cpp

```diff
@@ -688,15 +688,42 @@ comp_selection<item_comp> player::select_item_component( const std::vector<item_
         uimenu cmenu;
         // Populate options with the names of the items
         for( auto &map_ha : map_has ) {
-            std::string tmpStr = item::nname( map_ha.type ) + _( " (nearby)" );
+            std::string tmpStr;
+            if( item::count_by_charges( map_ha.type ) ) {
+                tmpStr = string_format( _( "%s (%d charges nearby)" ),
+                                        item::nname( map_ha.type ),
+                                        map_inv.charges_of( map_ha.type ) );
+            } else {
+                tmpStr = string_format( _( "%s (%d nearby)" ),
+                                        item::nname( map_ha.type ),
+                                        map_inv.amount_of( map_ha.type ) );
+            }
             cmenu.addentry( tmpStr );
         }
         for( auto &player_ha : player_has ) {
-            cmenu.addentry( item::nname( player_ha.type ) );
+            std::string tmpStr;
+            if( item::count_by_charges( player_ha.type ) ) {
+                tmpStr = string_format( _( "%s (%d charges on person)" ),
+                                        item::nname( player_ha.type ),
+                                        charges_of( player_ha.type ) );
+            } else {
+                tmpStr = string_format( _( "%s (%d on person)" ),
+                                        item::nname( player_ha.type ),
+                                        amount_of( player_ha.type ) );
+            }
+            cmenu.addentry( tmpStr );
         }
         for( auto &elem : mixed ) {
-            std::string tmpStr = item::nname( elem.type ) + _( " (on person & nearby)" );
-            cmenu.addentry( tmpStr );
+            std::string tmpStr;
+            if( item::count_by_charges( elem.type ) ) {
+                tmpStr = string_format( _( "%s (%d charges nearby & on person)" ),
+                                        item::nname( elem.type ),
+                                        map_inv.charges_of( elem.type ) + charges_of( elem.type ) );
+            } else {
+                tmpStr = string_format( _( "%s (%d nearby & on person)" ),
+                                        item::nname( elem.type ),
+                                        map_inv.amount_of( elem.type ) + amount_of( elem.type ) );
+            }
```

Comment:
you forgot to put `cmenu.addentry` back in.

Reply:
No it's right there, one line below.